### ISSUE TITLE
Create CheckstyleFile class to collect all information on file under processing

### ISSUE THREAD

**Issue by MEZk** [checkstyle/checkstyle#4527]
This issue is blocked by https://github.com/checkstyle/checkstyle/issues/4523 .

Followed by https://github.com/checkstyle/checkstyle/pull/4468#discussion_r123883389
 
We need to create a single class like CheckstyleFile. It will contain the file name, FileContents, and everything else related to the input and output of the [process method](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheck.java#L73).

**Comment by rnveach** [comment]
> Why do we need to include the output of the process method into CheckstyleFile? 

My original thought to include this was because it is all related to the file being worked on.
`CheckstyleFile` is named like a high level class, so it should contain everything on the file throughout the entire lifecycle. So why not the results too?
`FileProcessingResult` requires `fileText`, and we discussed adding file name to it in PR.  As discussed above, `CheckstyleFile` will also contain these same fields. It seems like the 2 classes will have an overlap of the same data.
Hence my reason for including the results into the class.

> FileProcessingResult represents results ... It can include CheckstyleFile which can be an argument

If we don't want to pass `FileProcessingResult` to all the FileSets and keep it as only a return of `processFile`, than I am ok with it. We should still not duplicate fields between the 2, especially if one is going to be a field of the other.

**Comment by romani** [comment]
> We need to create a single class like CheckstyleFile. It will contain the file name, FileContents, and everything else related to the input and output

please share with me a reason to keep FileText/Filecontent together with output/violations .

**Comment by MEZk** [comment]
@romani 
> please share with me a reason to keep FileText/Filecontent together with output/violations .

It was my question to @rnveach . He explained the reason https://github.com/checkstyle/checkstyle/issues/4527#issuecomment-311124115

**Comment by MEZk** [comment]
As @romani decided to exclude https://github.com/checkstyle/checkstyle/issues/4523 from GSoC this issue should be postponed till https://github.com/checkstyle/checkstyle/issues/4523 too.
